test/membrane_rtc_engine/tee_test.exs

```diff
@@ -241,6 +243,66 @@ defmodule Membrane.RTC.Engine.TeeTest do
     Pipeline.terminate(pipeline, blocking?: true)
   end
 
+  test "forwards VoiceActivityChanged" do
+    track = build_h264_track()
+    pipeline = build_video_pipeline(track, [])
+
+    Enum.each(track.variants, &mark_variant_as_resumed(pipeline, &1))
+
+    Enum.each(track.variants, fn variant ->
+      assert_sink_event(pipeline, :sink, %TrackVariantResumed{variant: ^variant})
+    end)
+
+    request_track_variant(pipeline, :high)
+
+    buffer = %Buffer{payload: <<>>, metadata: %{is_keyframe: true}}
+    send_buffer(pipeline, :high, buffer)
+
+    # Send vad events on :high and low
+    event = %VoiceActivityChanged{voice_activity: :speech}
+
+    Pipeline.execute_actions(pipeline,
+      forward: {{:source, :high}, {:execute_actions, event: {:output, event}}}
+    )
+
+    # We're already on high, so we expect the event to be forwarded
+    assert_sink_event(pipeline, :sink, ^event)
+
+    Pipeline.terminate(pipeline, blocking?: true)
+  end
+
+  test "forwards VoiceActivityChanged after TrackVariantSwitched" do
+    track = build_h264_track()
+    pipeline = build_video_pipeline(track, [])
```

Comment:
I am not convinced to use H264 track and video pipeline for testing VoiceActivityChanged. I think it should be possible to send `VoiceActivityChanged` only to a Tee that routes an audio track. In the other case we should raise :/

Reply:
> If we're introducing track type awareness, we could have skipped the entire forward event on TrackVariantSwitched as simulcast is not possible for audio disappointed

TrackVariantSwitched is always sent regardless of simulcast being enabled or disabled. Take a look here https://hexdocs.pm/membrane_rtc_engine/track_lifecycle.html#content

> Also, I don't have a problem sending VoiceActivityChanged on video tracks. I see this as totally viable. Firstly, someone could theoretically want to use this event to indicate that the person is talking ai. their lips are moving and it would be okay. Also, we had some instances of community members dealing with accessibility. In this case, speech isn't necessarily limited to audio and you might want to use it on a sign language interpreter video or something like that.

We don't have any other endpoints that can publish tracks so to send the new event to indicate one of the situations you described someone would have to write its own endpoint that publishes tracks. While it's possible I think we should limit the usage of this event until we have very specific use case to handle.